### PR TITLE
[DROOLS-3905] Duplicating an instance causes unexpected error

### PR DESCRIPTION
**Pull request by yesamer**
@kkufova @gitgabrio @danielezonca @jomarko Can you please review it?

https://issues.jboss.org/browse/DROOLS-3905

PR Related:https://github.com/kiegroup/drools-wb/pull/1158

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.Test;
+
+import static org.drools.scenariosimulation.api.model.FactIdentifier.DESCRIPTION;
+import static org.junit.Assert.assertEquals;
+
+public class FactIdentifierTest {
+
+    @Test
+    public void getClassNameWithoutPackage() {
+        FactIdentifier factIdentifier = DESCRIPTION;
```
Comment (danielezonca) [inline drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java]: I think the test could be more reliable using a new `FactIdentifier`. If `DESCRIPTION` className will change for any reason, this test will fail too. Wdyt?

File: drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Test
+    public void getClassNameWithoutPackage_NoPackage() {
+        FactIdentifier factIdentifier = new FactIdentifier("test", "Test");
+        assertEquals(factIdentifier.getClassNameWithoutPackage(), "Test");
+    }
+
+    @Test
+    public void getPackageWithoutClassName() {
+        FactIdentifier factIdentifier = DESCRIPTION;
```
Comment (danielezonca) [inline drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java]: Same as above

**Review by danielezonca — APPROVED** [review]
Approved with a couple of optional comments

**Review by kkufova — APPROVED** [review]
The fix works well. Thank you! Approving.

:+1:

File: drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java (hunk 1 above)
Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java]: Done!

File: drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java (hunk 2 above)
Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/test/java/org/drools/scenariosimulation/api/model/FactIdentifierTest.java]: Done!

**Review by gitgabrio — APPROVED** [review]
@yesamer 
Fine for me

**Comment by yesamer** [comment]
@kiegroup/gatekeepers Hi! Can you please merge it?

**Pull request merged**